packages/ui/src/components/SSelect/SSelectBase.vue

```diff
@@ -0,0 +1,150 @@
+<script setup lang="ts">
+import { SSelectSize, Option } from './types'
+import { computed, provide, readonly } from 'vue'
+import { useSelectModel } from './tools'
+import { SSelectApi, S_SELECT_API_KEY } from './api'
+import { templateRef, useVModel, useToggle, onClickOutside, unrefElement, whenever, and } from '@vueuse/core'
+import { usePopper } from '@/composables/popper'
+import { useConditionalScope } from '@/composables/conditional-scope'
+
+const props = withDefaults(
+  defineProps<{
+    disabled?: boolean
+    multiple?: boolean
+
+    modelValue?: any
+    options?: Option[]
+
+    size?: SSelectSize
+
+    /**
+     * This value is used if `label` slot is missing
+     */
+    label?: string
+
+    /**
+     * TODO
+     *
+     * - Doesn't allow to unselect value in single mode
+     * - Doesn't allow to unselect last the only one picked value in multiple mode
+     * - Picks some value automatically (the first one) if `modelValue` is null and there are some available options
+     */
+    mandatory?: boolean
+
+    /**
+     * @default 's-select-dropdown-transition'
+     */
+    dropdownTransitionName?: string
+
+    /**
+     * TODO
+     */
+    syncMenuAndInputWidths?: boolean
+  }>(),
+  {
+    size: SSelectSize.Md,
+    options: () => [],
+    modelValue: null,
+    multiple: false,
+    disabled: false,
+    dropdownTransitionName: 's-select-dropdown-transition',
+    syncMenuAndInputWidths: false,
+  },
+)
+
+const emit = defineEmits<(event: 'update:modelValue', value: any) => void>()
+
+const model = useVModel(props, 'modelValue', emit)
+const multiple = computed<boolean>(() => props.multiple)
+const disabled = computed<boolean>(() => props.disabled)
+const options = computed<Option[]>(() => props.options)
+const size = computed(() => props.size)
+const label = computed<string | null>(() => props.label ?? null)
+
+const modeling = useSelectModel({
+  model,
+  multiple,
+  options,
+})
+
+const popperRef = templateRef<HTMLDivElement | null>('popperRef')
+const popper = templateRef<HTMLDivElement | null>('popper')
+
+const { active: popperActive } = usePopper({
+  popperElem: popper,
+  referenceElem: popperRef,
+  placement: 'bottom-start',
+  active: true,
+  distance: 4,
+})
+
+const [showPopper, togglePopper] = useToggle(false)
+
+useConditionalScope(showPopper, () => {
+  onClickOutside(
+    popper,
+    (e) => {
+      const isClickInsideOfControlSlot = e.composedPath().includes(unrefElement(popperRef) as any)
+      // console.log('click outside:', e, { isClickInsideOfControlSlot })
```

Comment:
Could you remove commented console.log please?

Reply:
Sure